### PR TITLE
fix: avatar not changed in tooltip when updating avatar

### PR DESCRIPTION
**Pull request by dummy-1111**
### Fixed Issues
$ https://github.com/Expensify/App/issues/20672
PROPOSAL: https://github.com/Expensify/App/issues/20672#issuecomment-1589597755

### Tests
1. Go to offline mode
2. Go to Settings -> Profile and change the avatar
3. Go to any chat and hover on email to see user details in tooltip
4. Verify if updated avatar is displayed
5. Go to online mode and take steps 2 - 4.

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go to Settings -> Profile and change the avatar
2. Go to any chat and hover on email to see user details in tooltip
3. Verify if updated avatar is displayed

### QA Steps
1. Go to offline mode
2. Go to Settings -> Profile and change the avatar
3. Go to any chat and hover on email to see user details in tooltip
4. Verify if updated avatar is displayed
5. Go to online mode and take steps 2 - 4.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/126839717/aae457a0-4390-488d-b474-18f987cafbe6

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/126839717/07441d91-3a5e-4b5b-b4df-c9a49d44d903

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/126839717/a8568a06-3c5b-40f4-aef5-3b1fe8632e83

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/126839717/9b0702fa-1f42-4126-8fe5-b14379faf3ef

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/126839717/c7bb686f-0dde-4737-8c23-7eae00f091a2

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/126839717/cb5a555c-e627-4505-a29e-e757c50f5308

</details>

### CODE REVIEW

**Comment by pecanoro** [comment]
Reassigning since @allroundexperts can review it sooner!

File: src/libs/actions/PersonalDetails.js [hunk 1]
```diff
@@ -416,7 +454,7 @@ function updateAvatar(file) {
  */
 function deleteAvatar() {
     // We want to use the old dot avatar here as this affects both platforms.
-    const defaultAvatar = UserUtils.getDefaultAvatarURL(currentUserEmail);
+    const defaultAvatar = UserUtils.getDefaultAvatarURL(currentUserEmail, true);
```
[the remark this answers is not in the document]
Comment (s-alves10) [reply src/libs/actions/PersonalDetails.js:457]: If set to false, default user avatar URL is different than the ones that are current using. When user removes avatar, there is an inconsistency. You can easily check that.

Please let me know if you still want to change it

**Comment by s-alves10** [comment]
> @s-alves10 Can you also check why the uploaded image while in offline mode, does not show up in the image viewer modal?

This seems out of scope but let me check.

File: src/libs/actions/PersonalDetails.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (s-alves10) [reply src/libs/actions/PersonalDetails.js:461]: Other avatars get avatar URL with the below function.

**Comment by s-alves10** [comment]
When you upload an image in offline mode, avatar is stored as blob. blob url hasn't image extension in it and image check fails here

https://github.com/Expensify/App/blob/0a471b016d9b36c986a81680ef2518d2ae749937/src/components/AttachmentView.js#L110

That's why it isn't displayed as image

**Comment by s-alves10** [comment]
We might need to handle blob urls for offline mode

**Comment by s-alves10** [comment]
Do you think this is a regression?

**Comment by s-alves10** [comment]
@allroundexperts 

We have now similar bugs in ND.
1. In offline mode, ImageView in profile page doesn't work correctly as you mentioned
2. If you upload new avatar in offline status and go to online, avatars doesn't work correctly

These bugs are currently in ND and not caused by this PR
I don't think these are regression. Can you tell me your thoughts?

**Comment by s-alves10** [comment]
@mountiny 

Those bugs are not related to this PR. I found them on staging

**Comment by s-alves10** [comment]
> @s-alves10 Can you please share steps to reproduce [this](https://github.com/Expensify/App/pull/20713#issuecomment-1590362342) on staging?

@allroundexperts 
Issue 1

https://github.com/Expensify/App/assets/126839717/40915cec-3ca9-4928-a135-52a8352bd754

Issue 2
https://github.com/Expensify/App/assets/126839717/367a36b8-2870-483f-b635-7560e32432d6

**Comment by s-alves10** [comment]
Issue 1 is related to `AttachView` component. It can't show blob image. This can be addressed in a separate issue, I think

**Comment by mountiny** [comment]
I think we should fix this properly also with the new issue this introduces.

**Comment by s-alves10** [comment]
> I think we should fix this properly also with the new issue this introduces.

Excuse me, but can you explain in more detail? There are no issues this PR newly introduces.

**Comment by pecanoro** [comment]
This is the [issue](https://github.com/Expensify/App/pull/20713#issuecomment-1590447359) with the profile, right? Would it be that hard to fix in this PR as well?

**Comment by mountiny** [comment]
Maybe we got a bit lost in many comments here

@allroundexperts would you be able to summarize the problem and current situation we got here now, thanks?

File: src/libs/actions/PersonalDetails.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (s-alves10) [reply src/libs/actions/PersonalDetails.js]: Updated

**Comment by s-alves10** [comment]
> @s-alves10 Can you verify this as well?

Is this one I have to check?

**Comment by s-alves10** [comment]
Checked.

**Comment by mountiny** [comment]
@allroundexperts @s-alves10 can you confirm that once you go back online, the avatar will show correctly in [this scenario](https://github.com/Expensify/App/pull/20713#issuecomment-1590362342)?

**Review by allroundexperts — APPROVED** [review]
Looks good. Tests well as well!

**Comment by mountiny** [comment]
@pecanoro  I think its good to CP this and make a follow up issue for the problem discovered by this PR. All yours 🙇

**Comment by pecanoro** [comment]
Awesome, doing a final test run!

**Comment by pecanoro** [comment]
The bug #1 we found with the profile image, I think it might be working on dev and some fix might have been recently merged as it works well for me. We will see after the back-end deploy anyways, otherwise, all good! Approving!

**Review by pecanoro — APPROVED** [review]

**Comment by s-alves10** [comment]
> @pecanoro I think its good to CP this and make a follow up issue for the problem discovered by this PR. All yours 🙇

I can work on the follow up issue. Thanks

**Comment by pecanoro** [comment]
Oh shit, I forgot to add the CP label.

**Comment by pecanoro** [comment]
Tested on staging, bug was fixed!

**Pull request merged**